mowgli_etl/pipeline/wdc/parsimonious_parser/wdc_parsimonious_dimension_parser.py

```diff
@@ -32,27 +37,49 @@ def parse(self, *, entry: WdcOffersCorpusEntry):
         return_flag = False
         __VISITOR = WdcParsimoniousNodeVisitor()
 
-        def __generate_dimensions():
-            retVal = WdcProductDimensions.from_dict(dataclasses.asdict(__VISITOR.dictionary))
-            __VISITOR.reset()
-            if retVal.width.value is None and retVal.length.value is None and retVal.depth.value is None and retVal.height.value is None:
+        def __generate_dimensions(source):
+            __VISITOR = WdcParsimoniousNodeVisitor()
+            __VISITOR.visit(source)
+            retVal = WdcProductDimensions.from_dict(
+                dataclasses.asdict(__VISITOR.dictionary)
+            )
+            if (
+                retVal.width is None
+                and retVal.length is None
+                and retVal.depth is None
+                and retVal.height is None
+                and retVal.weight is None
+                and retVal.power is None
+            ):
                 return None
             return retVal
 
-        if entry.KeyValuePairs is not None:
-            if "dimensions" in entry.KeyValuePairs.keys():
-                KeyValuePairs = self.__GRAMMAR.parse(entry.KeyValuePairs["dimensions"])
-                __VISITOR.visit(KeyValuePairs)
-                result = __generate_dimensions()
+        if entry.key_value_pairs is not None:
+            if "dimensions" in entry.key_value_pairs.keys():
+                key_value_pairs = self.__GRAMMAR.parse(
+                    entry.key_value_pairs["dimensions"]
+                )
+                result = __generate_dimensions(key_value_pairs)
                 if result:
-                    returns.append((result, 'KeyValuePairs'))
+                    returns.append((result, "key_value_pairs", entry.key_value_pairs))
+
+            if "weight" in entry.key_value_pairs.keys():
+                key_value_pairs = self.__GRAMMAR.parse(entry.key_value_pairs["weight"])
+                result = __generate_dimensions(key_value_pairs)
+                if result:
+                    returns.append((result, "key_value_pairs", entry.key_value_pairs))
 
         if entry.description is not None:
             description = self.__GRAMMAR.parse(entry.description)
-            __VISITOR.visit(description)
-            result = __generate_dimensions()
+            result = __generate_dimensions(description)
+            if result:
+                returns.append((result, "description", entry.description))
+
+        if entry.spec_table_content is not None:
+            spec_table_content = self.__GRAMMAR.parse(entry.spec_table_content)
+            result = __generate_dimensions(spec_table_content)
             if result:
-                returns.append((result, "description"))
+                returns.append((result, "spec_table_content", entry.spec_table_content))
```

Comment:
Use a `@dataclass` or NamedTuple as the "returns" element type rather than a positional tuple.